explorer/chartscache.go

```diff
@@ -0,0 +1,403 @@
+				break
+			}
+		}
+		cacheData[dbtypes.TicketsByBlocks].Height = cacheData[dbtypes.TicketsByBlocks].Height[:index]
+		cacheData[dbtypes.TicketsByBlocks].Solo = cacheData[dbtypes.TicketsByBlocks].Solo[:index]
+		cacheData[dbtypes.TicketsByBlocks].Pooled = cacheData[dbtypes.TicketsByBlocks].Pooled[:index]
+	}
+
+	// "ticket-spend-type" chart data.
+	data, ok = cacheData[dbtypes.TicketSpendT]
+	if ok {
+		var index = len(data.Height) - 1
+		for ; index > 0; index-- {
+			if data.Height[index] < heightAfterAncestor {
+				break
+			}
+		}
+		cacheData[dbtypes.TicketSpendT].Height = cacheData[dbtypes.TicketSpendT].Height[:index]
+		cacheData[dbtypes.TicketSpendT].Unspent = cacheData[dbtypes.TicketSpendT].Unspent[:index]
+		cacheData[dbtypes.TicketSpendT].Revoked = cacheData[dbtypes.TicketSpendT].Revoked[:index]
+	}
+
+	// "tx-per-block" chart data.
+	data, ok = cacheData[dbtypes.TxPerBlock]
+	if ok {
+		var index = len(data.Height) - 1
+		for ; index > 0; index-- {
+			if data.Height[index] < heightAfterAncestor {
+				break
+			}
+		}
+		cacheData[dbtypes.TxPerBlock].Height = cacheData[dbtypes.TxPerBlock].Height[:index]
+		cacheData[dbtypes.TxPerBlock].Count = cacheData[dbtypes.TxPerBlock].Count[:index]
+	}
+
+	// "tx-per-day" chart data.
+	data, ok = cacheData[dbtypes.TxPerDay]
+	if ok {
+		var index = len(data.Time) - 1
+		for ; index > 0; index-- {
+			if data.Time[index].UNIX() < timeAfterAncestor {
+				break
+			}
+		}
+		cacheData[dbtypes.TxPerDay].Time = cacheData[dbtypes.TxPerDay].Time[:index]
+		cacheData[dbtypes.TxPerDay].Count = cacheData[dbtypes.TxPerDay].Count[:index]
+	}
+
+	// "fee-per-block" chart data.
+	data, ok = cacheData[dbtypes.FeePerBlock]
+	if ok {
+		var index = len(data.Height) - 1
+		for ; index > 0; index-- {
+			if data.Height[index] < heightAfterAncestor {
+				break
+			}
+		}
+		cacheData[dbtypes.FeePerBlock].Height = cacheData[dbtypes.FeePerBlock].Height[:index]
+		cacheData[dbtypes.FeePerBlock].SizeF = cacheData[dbtypes.FeePerBlock].SizeF[:index]
+	}
+
+	// "ticket-pool-size" chart data.
+	data, ok = cacheData[dbtypes.TicketPoolSize]
+	if ok {
+		var index = len(data.Time) - 1
+		for ; index > 0; index-- {
+			if data.Time[index].UNIX() < timeAfterAncestor {
+				break
+			}
+		}
+		cacheData[dbtypes.TicketPoolSize].Time = cacheData[dbtypes.TicketPoolSize].Time[:index]
+		cacheData[dbtypes.TicketPoolSize].SizeF = cacheData[dbtypes.TicketPoolSize].SizeF[:index]
+	}
+
+	// "ticket-pool-value" chart data.
+	data, ok = cacheData[dbtypes.TicketPoolValue]
+	if ok {
+		var index = len(data.Time) - 1
+		for ; index > 0; index-- {
+			if data.Time[index].UNIX() < timeAfterAncestor {
+				break
+			}
+		}
+		cacheData[dbtypes.TicketPoolValue].Time = cacheData[dbtypes.TicketPoolValue].Time[:index]
+		cacheData[dbtypes.TicketPoolValue].ValueF = cacheData[dbtypes.TicketPoolValue].ValueF[:index]
+	}
+
+	// Update the modified cache data.
+	cacheChartsData.Update(exp.Height(), cacheData)
```

Comment:
All the above modifications to the charts data already affect the stored `Data` map. Note that in this function above the map itself is not modified at all, just the fields of the stored structs, so it's not clear why reassigning `cacheData` to the `Data` field is done.  But even if the `cacheData` map were modified (insert/delete), those changes would still be visible in the already-stored map in `Data` because an initialized `map` is a reference type in golang.  In fact, internally, a map is a pointer.